### ISSUE TITLE
View Components are not supported

### ISSUE THREAD

**Issue by plneple** [mrousavy/nitro#425]
### How were you trying to build the app?

Hey @mrousavy,

Is there any discussion or roadmap about providing codegen support for View Components ?

What would the minimum viable implementation have to support ?

### Full build logs

```tsx
I am making a build error report, because there is no better template.
```

### Project dependencies

```json
-
```

### Nitro Modules Version

latest

### Target platforms

iOS, Android

### Operating system

MacOS, Windows, Linux

### Can you build the Nitro Modules Example app?

Yes, I can successfully build the Example app here

### Additional information

- [ ] I am using Expo
- [X] I am using Nitrogen (nitro-codegen)
- [X] I have read and followed the [Troubleshooting Guide](https://nitro.margelo.com/docs/troubleshooting).
- [ ] I created a reproduction PR to reproduce this issue here in the nitro repo. (See [Contributing](https://nitro.margelo.com/docs/contributing) for more information)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/nitro/issues) and found none.

**Comment by mrousavy** [comment]
View components would work with the workaround mentioned in the docs.

We're working on first class support tho which will have better flexibility, better performance and better DX. No timeline, but this will likely take a few more days or weeks.

**Comment by plneple** [comment]
Is there any way to contribute ?

I have been fiddling around with TurboModules with just the problem described in the docs.
`RawProps` seems to hold everything that is required for proper implementation (raw `jsi::Value` for props and `jsi::Runtime`, but the properties are private) so one cheecky typecast could maybe be a solution ?

However, the documentation mentions (and it in fact exists in `react-native` codebase) constructing the type using `folly:dynamic` as data.

I was just wondering if that is a solution you would consider ?

**Comment by hannojg** [comment]
Hey, we are exactly working on a solution for this, actively changing react-native core. I think so far it's a bit hard to contribute to that in some way (as we have to wait for the react native changes to be released in a new version).
You can play around with the state in the linked PR, and maybe try out the patches locally for yourself:

- https://github.com/mrousavy/nitro/pull/386

However, the PR currently only provides the changes needed to bring the infrastructure to rn core. We are still working on an API for Nitro, which as Marc said, will take a bit more time.

**Comment by mrousavy** [comment]
> However, the documentation mentions (and it in fact exists in `react-native` codebase) constructing the type using `folly:dynamic` as data.
> 
> I was just wondering if that is a solution you would consider ?

Nope - `folly::dynamic` is a workaround and not a fast solution as we have to add another complex layer. Also, it doesn't support all of Nitro's types.

As @hannojg said, we are changing some stuff in RN Core to support this and it seems to be working fine :)

**Comment by plneple** [comment]
Yeah, I meant react-native was using folly::dynamic, not suggesting it for NitroModules.

Thank you for the answer. I will be following the linked PR.

**Comment by mrousavy** [comment]
Nitro Views are now released in 0.23.0! 🥳  https://nitro.margelo.com/docs/view-components
